data-prepper-core/src/main/java/org/opensearch/dataprepper/peerforwarder/RemotePeerForwarder.java

```diff
@@ -98,22 +101,13 @@ public Collection<Record<Event>> forwardRecords(final Collection<Record<Event>>
                 recordsToBeProcessedLocallyCounter.increment(entry.getValue().size());
             } else {
                 recordsToBeForwardedCounter.increment(entry.getValue().size());
-                AggregatedHttpResponse httpResponse;
                 try {
-                    httpResponse = peerForwarderClient.serializeRecordsAndSendHttpRequest(entry.getValue(),
-                            destinationIp, pluginId, pipelineName);
+                    submitForwardingRequest(entry.getValue(), destinationIp);
                 } catch (final Exception ex) {
-                    httpResponse = null;
                     LOG.warn("Unable to send request to peer, processing locally.", ex);
-                }
-
-                if (httpResponse == null || httpResponse.status() != HttpStatus.OK) {
                     recordsToProcessLocally.addAll(entry.getValue());
```

Comment:
This shouldn't happen here. You are adding the failed records to buffer on line 197 and also returning them which makes the failed records to be processed twice.

Since we are making async call to forward records, we should return an empty collection here and let the buffer handle failed records